### PR TITLE
Using env variables to control execution parameters

### LINKED ISSUE #75: Allow control from environment variables
As a way to be as less intrusive as possible and yet flexible, introduce use of os.environ.get() for most of the parameters of the code, so that one can make use of them while running headless tasks in a more flexible way.

As a rule of thumb, they would be named "ds_" and following the variable name. Default values stays the same. This will be used in the PR for issue #72

### PR DESCRIPTION
**Pull request by lissyx**
Fixes #75

### CODE REVIEW

**Comment by lissyx** [comment]
That one should be good to review too @kdavis-mozilla

File: DeepSpeech.ipynb [hunk 1]
```diff
@@ -1280,8 +1285,8 @@
    "source": [
     "# Define CPU as device on which the muti-gpu training is orchestrated\n",
     "with tf.device('/cpu:0'):\n",
-    "    # Obtain ted lium data\n",
-    "    ted_lium = read_data_sets(tf.get_default_graph(), './data/ted', batch_size, n_input, n_context)\n",
+    "    # Obtain all the data, defaulting to TED LIUM\n",
+    "    run_data_set = ds_importer.read_data_sets(tf.get_default_graph(), os.environ.get('ds_dataset_path', './data/ted'), batch_size, n_input, n_context)\n",
```
Comment (kdavis-mozilla) [inline DeepSpeech.ipynb]: nit, could you change the variable name `run_data_set` to `data_sets`to be consistent with `train`'s naming and with the fact that it's more than one data set.

File: DeepSpeech.ipynb [hunk 2]
```diff
@@ -1371,9 +1376,9 @@
     "            'n_hidden_6': n_hidden_6, \\\n",
     "            'n_cell_dim': n_cell_dim, \\\n",
     "            'n_character': n_character, \\\n",
-    "            'total_batches_train': ted_lium.train.total_batches, \\\n",
-    "            'total_batches_validation': ted_lium.dev.total_batches, \\\n",
-    "            'total_batches_test': ted_lium.test.total_batches \\\n",
+    "            'total_batches_train': run_data_set.train.total_batches, \\\n",
+    "            'total_batches_validation': run_data_set.dev.total_batches, \\\n",
+    "            'total_batches_test': run_data_set.test.total_batches \\\n",
```
Comment (kdavis-mozilla) [inline DeepSpeech.ipynb]: As we are now switching data sets, we should log which data set the data is from. Something like
```python
...
        'data_set': { \
            'name': os.environ.get('ds_importer', 'ted_lium')
        }, \
...
```
Also, I'd guess [doyouunderstand.me](http://doyouunderstand.me) should now display the data set's name too.

Comment (lissyx) [inline DeepSpeech.ipynb]: Of course, I think we should take care of this once we have everything running properly end-to-end.

**Comment by lissyx** [comment]
All comments should be addressed :) @kdavis-mozilla

**Comment by kdavis-mozilla** [comment]
LGTM

**Pull request merged**